farm/data_handler/dataset.py

```diff
@@ -18,14 +23,30 @@ def convert_features_to_dataset(features):
     tensor_names = list(features[0].keys())
     all_tensors = []
     for t_name in tensor_names:
-        try:
-            cur_tensor = torch.tensor(
-                [sample[t_name] for sample in features], dtype=torch.long
-            )
-        except ValueError:
-            cur_tensor = torch.tensor(
-                [sample[t_name] for sample in features], dtype=torch.float32
-            )
+        # Conversion of floats
+        if t_name == 'regression_label_ids':
+            cur_tensor = torch.tensor([sample[t_name] for sample in features], dtype=torch.float32)
+        else:
+            try:
+                # Checking weather a non-integer will be silently converted to torch.long
+                if isinstance(features[0][t_name], numbers.Number):
+                    base = features[0][t_name]
+                elif isinstance(features[0][t_name], list):
+                    if len(features[0][t_name]) > 0:
+                        base = features[0][t_name][0]
+                    else:
+                        base = 1
+                else:
+                    base = features[0][t_name].ravel()[0]
+                if not np.issubdtype(type(base), np.integer):
+                    logger.warning(f"Problem during conversion to torch tensors:\n"
+                                   f"A non-integer value for '{t_name}' with a value of: "
+                                   f"'{base}' will be converted to a torch tensor of dtype long.")
+            except:
+                logger.warning("conversion checks did not work")
```

Comment:
Let's make this warning here more descriptive. If users see this it should be clear where this comes from and what this means, e.g "Could not determine type for feature 'XXX'. Converting now to a tensor of default type long."